dbt/include/athena/macros/materializations/snapshots/snapshot.sql

```diff
@@ -1,5 +1,322 @@
+              AS is_current_record {%- if not loop.last -%},{%- endif -%}
+              {% else %}
+                {{ column.name }} {%- if not loop.last -%},{%- endif -%}
+              {% endif %}
+        {% endfor %}
+        ,{{ current_timestamp() }} AS dbt_snapshot_at
+      from {{ target }}
+      WHERE dbt_unique_key IN ( SELECT dbt_unique_key FROM {{ source }} )
+      UNION ALL
+      SELECT
+        {% for column in source_columns %}
+            {{ column.name }} {%- if not loop.last -%},{%- endif -%}
+        {% endfor %}
+        ,{{ current_timestamp() }} AS dbt_snapshot_at
+      FROM {{ source }};
+    {%- endset -%}
+
+    {% call statement('create_new_snapshot_table') %}
+        {{ create_table_as(False, target_relation, sql) }}
+    {% endcall %}
+
+    {% do return(target_relation) %}
+{% endmacro %}
+
+{% materialization snapshot, adapter='athena' %}
+  {%- set config = model['config'] -%}
+
+  {%- set target_table = model.get('alias', model.get('name')) -%}
+
+  {%- set strategy_name = config.get('strategy') -%}
+  {%- set unique_key = config.get('unique_key') %}
+  {%- set file_format = config.get('file_format', 'parquet') -%}
+
+  {% set target_relation_exists, target_relation = get_or_create_relation(
+          database=none,
+          schema=model.schema,
+          identifier=target_table,
+          type='table') -%}
+
+
+  {% if not adapter.check_schema_exists(model.database, model.schema) %}
+    {% do create_schema(model.database, model.schema) %}
+  {% endif %}
+
+  {%- if not target_relation.is_table -%}
+    {% do exceptions.relation_wrong_type(target_relation, 'table') %}
+  {%- endif -%}
+
+  {{ run_hooks(pre_hooks, inside_transaction=False) }}
+
+  {% set strategy_macro = strategy_dispatch(strategy_name) %}
+  {% set strategy = strategy_macro(model, "snapshotted_data", "source_data", config, target_relation_exists) %}
+
+  {% if not target_relation_exists %}
+
+      {% set build_sql = build_snapshot_table(strategy, model['compiled_sql']) %}
+      {% set final_sql = create_table_as(False, target_relation, build_sql) %}
+
+  {% else %}
+
+      {{ adapter.valid_snapshot_target(target_relation) }}
+
+      {% set staging_table = athena__build_snapshot_staging_table(strategy, sql, target_relation) %}
+
+      {% set missing_columns = adapter.get_missing_columns(staging_table, target_relation) %}
+
+      {% do create_columns(target_relation, missing_columns) %}
+
+      {% set new_snapshot_table = athena__create_new_snapshot_table(target = target_relation, source = staging_table) %}
+
+      {% set final_sql = athena__snapshot_merge_sql(
+            target = target_relation,
+            source = new_snapshot_table
+         )
+      %}
+
+  {% endif %}
+
+  {% call statement('main') %}
+      {{ final_sql }}
+  {% endcall %}
+
+  {% do persist_docs(target_relation, model) %}
+
+  {{ run_hooks(post_hooks, inside_transaction=True) }}
+
+  {{ adapter.commit() }}
```

Comment:
what is the reasoning behind that? athena doesn't really support isolation levels. I think that we can safely exlude it

Reply:
Removed. I had referenced from another adapter but forgot to remove this.